cmd/internal/ec2/add.go

```diff
@@ -0,0 +1,174 @@
+	}
+	if strings.EqualFold(cu.Name, "root") {
+		return errors.Errorf("Please run the command again with user: root. \n")
+	}
+	return nil
+}
+
+func checkSystemdExists() error {
+	message := "Checking if systemd exists\n"
+	printer.Infof(message)
+	reportStep(message)
+
+	// _, serr := exec.LookPath("systemd")
+	// if serr != nil {
+	// 	return errors.Errorf("Could not find systemd binary in your OS.\n We don't have support for non-systemd OS as of now; For more information please contact observability-support@postman.com.\n")
+	// }
+	return nil
+}
+
+func configureSystemdFiles(collectionId string) error {
+	message := "Configuring systemd files\n"
+	printer.Infof(message)
+	reportStep(message)
+
+	// Write collectionId and postman-api-key to go template file
+
+	tmpl, err := template.ParseFS(envFileFS, envFileTemplateName)
+	if err != nil {
+		return errors.Errorf("failed to parse systemd env template file with error :%q\n Please contact observability-support@postman.com  with this log for further assistance.\n", err)
+	}
+
+	data := struct {
+		PostmanAPIKey string
+		CollectionId  string
+	}{
+		PostmanAPIKey: os.Getenv("POSTMAN_API_KEY"),
+		CollectionId:  collectionId,
+	}
+	envFile, err := os.Create("env-postman-lc-agent")
+	if err != nil {
+		return errors.Errorf("Failed to write values to env file with error %q", err)
+	}
+
+	err = tmpl.Execute(envFile, data)
+	if err != nil {
+		return errors.Errorf("Failed to write values to env file with error %q", err)
+	}
+
+	// Ensure /etc/default exists
+	cmd := exec.Command("mkdir", []string{"-p", envFileBasePath}...)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.Errorf("failed to find %s directory with err %q \n", envFileBasePath, err)
+	}
+
+	// move the file to /etc/default
+	cmd = exec.Command("mv", []string{"env-postman-lc-agent", envFilePath}...)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.Errorf("failed to create postman-lc-agent env file in /etc/default directory with err %q \n", err)
+	}
+
+	// Ensure /usr/lib/systemd/system exists
+	cmd = exec.Command("mkdir", []string{"-p", serviceFileBasePath}...)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.Errorf("failed to find %s directory with err %q \n", serviceFileBasePath, err)
+	}
+
+	err = os.WriteFile(serviceFilePath, []byte(serviceFile), 0600)
+	if err != nil {
+		return errors.Errorf("failed to create %s file in %s directory with err %q \n", serviceFileName, serviceFilePath, err)
+	}
+
+	return enablePostmanAgent()
+}
+
+// Starts the postman LCA agent as a systemd service
+func enablePostmanAgent() error {
+	reportStep("Enabling postman-lc-agent as a service")
+
+	cmd := exec.Command("systemctl", []string{"daemon-reload"}...)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Errorf("failed to run systemctl daeomon-reload %q \n", err)
+	}
+	// systemctl start postman-lc-service
+	cmd = exec.Command("systemctl", []string{"start", serviceFileName}...)
```

Comment:
We need "enable" so that it starts on the next boot.

I think that "enable --now" will also start it immediately